hazelcast/src/hazelcast/client/spi.cpp

```diff
@@ -840,7 +842,22 @@ namespace hazelcast {
                     member_list_snapshot result;
                     result.version = version;
                     for (auto &m : members) {
-                        result.members.insert({m.get_uuid(), m});
+                        auto const &address_map = m.address_map();
+                        if (address_map.empty()) {
+                            result.members.insert({m.get_uuid(), m});
+                        } else {
+                            auto found = address_map.find(CLIENT);
+                            address member_address;
+                            if (found != address_map.end()) {
+                                member_address = found->second;
+                            } else {
+                                found = address_map.find(MEMBER);
+                                assert(found != address_map.end());
+                                member_address = found->second;
+                            }
+                            member new_member(member_address, m.get_uuid(), m.is_lite_member(), m.get_attributes(), m.address_map());
```

Comment:
I realize there isn't a check for the `client_network_config::use_public_address` configuration here. So, the address of a member that is passed to a membership event handler can change with the new implementation. Isn't this a breaking change? There is already an address translation logic implemented in `ClientConnectionManagerImpl::translate`. Why is this also necessary?

Reply:
Our implementation follows the Java client reference impl. I believe the logic is this: If the use_public_address is not configured (i.e. it is false), use the address coming in the address_map if address_map is sent from the server. It is given higher priority than the server `address` field inside the member_info. Hence, if there is an address_map, the address field is ignored, and this is done regardless of the `use_public_address` configuration.